### ISSUE TITLE
@DebugTest annotation support

### ISSUE THREAD

**Issue by nagkumar** [junit-team/junit4#1475]
For Debugging Purpose, one may write many test cases.. which may not have asserts but may be required for breakpoints based debugging. 

It would be nice to have annotation @DebugTest in parallel to @Test. This is a clear indication that, such test code is meant for minimum viable tests to trace any exceptions in the main code

**Comment by kcooney** [comment]
What behavior would JUnit have for `@DebugTest` vs. `@Test`?

Is there a reason why you couldn't define your own `@DebugTest` annotation?

**Comment by kcooney** [comment]
@nagkumar closing for now. If you have time to answer my questions we could consider reopening.

**Comment by nagkumar** [comment]
Sorry, I missed your last update..
@DebugTest, can help is serration of test suites, specially when tests fail. It is mostly marker annotation. Also when such tests are run, it can automatically allow IDE debug integration and wait for debugging.

**Comment by stefanbirkner** [comment]
@nagkumar can you amplify "@debugtest, can help is serration of test suites, specially when tests fail." I don't understand. Maybe an example is helpful.

**Comment by nagkumar** [comment]
Stefan, when unit testing is a natural norm in mature product organisations, I have observed tonns of unit tests gets written. This way each feature may have many variations to same feature etc..

As the number of tests grow, many times, due to some change in xyz logic, if that xyz has 10 variations of tests all the 10 fail. In such context marking one of the 10 tests as @DebugTest would help to fix all others. Using each of these 10 variations with @Test would not convey new developers or even the same developer who has written them, which test he needs to focus and debug on priority basis. I have seen many people panic, if test shows 10 tests fail at a time..

Also, another area where I saw use of @DebugTest is to allow System.out.println/Logging as they are meant for easy debugging with debug logic readable to debugger.. specially for junior developers. I know many may not accept this style, but did feel the need to give way for junior developers to grow steadily

**Comment by kcooney** [comment]
I would prefer not to add marker annotations to JUnit. All JUnit annotations have behavior associated with them; adding a new annotation without behavior would be confusing. 

Luckily, It's very easy for projects to create their own annotations.

> Also when such tests are run, it can automatically allow
> IDE debug integration and wait for debugging.

I am not sure this is possible. In any case, if you can run your tests in an IDE them you can set breakpoints in any test method before you launch the tests.

**Comment by nagkumar** [comment]
Sure, thank you for noting/understanding  my observations.  I am also ok with the design decisions taken.
